autoload/neomake/makers/ft/d.vim

```diff
@@ -0,0 +1,50 @@
+" vim: ts=4 sw=4 et
+
+function! neomake#makers#ft#d#EnabledMakers()
+    if exists ('s:d_makers')
+        return s:d_makers
+    endif
+
+    let l:makers = []
+    if neomake#utils#Exists('dmd')
+        call add(l:makers,'dmd')
+    endif
+
+    let s:neomake_d_dub_include_dirs = ""
+    if !exists('g:neomake_d_nodub')
+        call s:UpdateDub()
+    endif
+    let s:d_makers = makers
+    return makers
+endfunction
+
+function! s:UpdateDub()
+    "Add dub directories
+    let l:tmp_file = findfile("dub.json", ".;")
+    if neomake#utils#Exists('dub') && !empty(l:tmp_file)
+        let l:tmp_dir = fnamemodify(l:tmp_file,':p:h')
+        let l:dub = eval(substitute(system("dub describe --annotate --root=" . tmp_dir)
+                    \ ,'\v[\n\r]','','g'))
+        let l:tmp_arr = []
+        for l:package in l:dub.packages
+            let l:path = l:package.path
+            for l:importPath in l:package.importPaths
+                call add(tmp_arr, l:path . l:importPath)
+            endfor
+        endfor
+        let s:neomake_d_dub_include_dirs = join(tmp_arr, ":")
```

Comment:
Could this value be returned instead? Then we could call this function in `neomake#makers#ft#d#dmd` to get a fresh result every time we build a new dmd maker (which is every time the user runs `:Neomake`).